### DISCUSSION TITLE
using deep track on M1 Mac

### DISCUSSION THREAD

**Discussion by ghost** [DeepTrackAI/DeepTrack2#122] [General]
Hi, first of all thank you so much for coming up with this amazing module. it very useful in my project in understanding bacterial driven systems.
I am still a beginner with this entire machine learning usage however I hope to pick things in coming days.

I had a question regarding the use of the codes on an M1 Mac as I errors and can't move ahead. this started when I shifted to this new system and with the old windows machine there were no issues.

is it because of the use of cupy for gpu accelerated processes as that is not supported by apple anymore.? or am I doing some else wrong? id really appreciate some help in this matter. 
Thanks again.

**Comment by BenjaminMidtvedt** [comment]
Hi! Our test suite does run on a Mac, so there should not be any fundamental incompatibilities. Could you let me know exactly what the errors are? DeepTrack is designed to run fine without cupy installed, see if cupy is not compatible with mac, it should not be a problem to leave it uninstalled

**Comment by ArthurCoet** [comment]
Hello,

I'm a PhD student in microbial oceanography, at MIO, Marseille.
I seem to have the same problem, once the code is launched, I get the following error: 

zsh: illegal hardware instruction python

I've seen that it doesn't need Cupy, but I can't get round this problem. Have you already run the code on MAC M1/M2?

Many thanks.

Arthur.

**Reply by BenjaminMidtvedt** [reply]
My bet is that the issue is with tensorflow, which has become very cumbersome lately for any os that is not linux.

Is the error when installing deeptrack or executing the code? If the latter, is it at the start of the script? If so, can you try running a simple script that is just:
```python
import tensorflow as tf
```

**Reply by ArthurCoet** [reply]
I agree with you i think it's because of tensorflow. There were no problems when installing deeptrack on the terminal, it's once I run the code on pycharm. 
I've just run the script you sent me and I still get the same error: 

zsh: illegal hardware instruction python

**Reply by BenjaminMidtvedt** [reply]
Ok, great! That confirms it. My bet is that it can be solved by downgrading tensorflow a version or two. I don't have a Mac myself so I can't verify it

**Reply by ArthurCoet** [reply]
So I did downgrade tensorflow (to version 2.5.0) and the line of code you gave me still doesn't works. Even when I try 01. deeptrack_introduction_tutorial.ipynb, I get the same error message as before. Maybe the dowgrade version is still too recent?

**Reply by ArthurCoet** [reply]
So i tried to downgrade even more Tensorflow, but i have still the same problem. I heard about a tensorflow adapt for MacOS which is "tensorflow Metal", have you ever heard of it ?
